SDWebImage/SDImageCache.m

```diff
@@ -248,7 +248,8 @@ - (NSData *)diskImageDataBySearchingAllPathsForKey:(NSString *)key {
         return data;
     }
 
-    for (NSString *path in self.customPaths) {
+    NSArray *customPaths = [self.customPaths copy];
```

Comment:
This is a kind of workaround, not a real fix. Correct fix is usage of the @synchronized or some dispatch queue ...